src/Workspaces/Core/Portable/EmbeddedLanguages/VirtualChars/VirtualCharSequence.Chunks.cs

```diff
@@ -67,9 +68,16 @@ public StringChunk(int firstVirtualCharPosition, string data)
             public override int Length => _underlyingData.Length;
 
             public override VirtualChar this[int index]
-                => new VirtualChar(
-                    _underlyingData[index],
-                    new TextSpan(_firstVirtualCharPosition + index, length: 1));
+            {
+                get
+                {
+                    var span = new TextSpan(_firstVirtualCharPosition + index, length: 1);
+                    var ch = _underlyingData[index];
+                    return char.IsSurrogate(ch)
+                        ? VirtualChar.Create(ch, span)
```

Comment:
❔ Is this the behavior we want for _paired_ surrogates?

Reply:
So, the way this works is we *always* conver the string into an `ArrayBuilder<VirtualChar>`.  We then only create a StringChunk if the number of VirtualChars matches the number of actual chars.  In the case of a *paired* surrogate , we could never have gotten here as that would have caused one less VirtualChar to be made, which would have meant we woudln't mak a StringChunk.

This is documented elsewhere.  but i'll add an assert here.